zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -382,4 +334,158 @@ private ExporterResourceProvider mockResourceProvider(
 
     return provider;
   }
+
+  @Nested
+  class ExportHandlerTests {
+
+    @TestTemplate
+    void shouldHandleExportedRecords(
+        final ExporterConfiguration config, final SearchClientAdapter clientAdapter)
+        throws IOException {
+      final var provider = new DefaultExporterResourceProvider();
+      provider.init(config);
+
+      final var exporter =
+          createExporter(
+              new HashSet<>(provider.getIndexDescriptors()),
+              new HashSet<>(provider.getIndexTemplateDescriptors()),
+              config);
+      for (final var handler : provider.getExportHandlers()) {
+        exportTest(exporter, handler, clientAdapter);
+      }
+    }
+
+    private <S extends ExporterEntity<S>, T extends RecordValue> Record<T> recordGenerator(
+        final ExportHandler<S, T> handler, final Supplier<Record<T>> createRecord) {
+      for (int i = 0; i < 500; i++) {
```

Comment:
❓ What is the reason behind this? Having here a loop?

Reply:
I should add a comment, but basically sometimes records are generated from the given `ValueType` but the handler cannot handle the record because it generates with the wrong intents

Reply:
That sounds like a good candidate fora flaky test then

Reply:
@EuroLew I think I understand now the issue better. The problem is that some handlers only support specific intents. Obviously we dont want to couple this knowledge of which intent is handled by which handler in our IT.

But I think it would be safer if we use here a while loop instead of a for. Otherwise there exist the chance we dont generate a fiting record.

What I think we could do is:

```
do {
// generate record
} while(!handler.handles(record))
```

Does this make sense?

Reply:
To even make this more performant (elegant?) we can move this one layer up in the set up code. Do it once for all handlers we have (generating a record + passing to the handler to get the expected entity) and then the test method only retrieves

- CamundaExporter
- ExpectedEntity
- ClientAdapter

Wdyt?

Reply:
I'm ok with passing the record it to stop it being generated twice per handler, but I think the record should still be passed in so there's a logical progression to the test in that 
1. generate the expected entity